### ISSUE TITLE
Preserve item/annotation link in notes after importing from Zotero RDF

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2540]
https://forums.zotero.org/discussion/comment/405292/#Comment_405292

Have to think about how to do this. I can't remember if the original item keys are in the output — if so, we should be able to rewrite the notes on import. If not, we could rewrite the notes on export to include the itemIDs or whatever identifiers we include in the output, and then rewrite again on import.

**Comment by AbeJellinek** [comment]
Zotero RDF exports include item IDs but not keys, so we'll need to rewrite on export.

Should we just do this as a general pre-processing step for exports and post-processing step for imports? I could be overthinking this, but I think preserving annotation links is more than Zotero RDF can handle right now, especially since it doesn't get any annotation data yet (progress on that in #2378 and zotero/translate#12). And there's no reason why other translators that export notes and attachments shouldn't preserve links too.

**Comment by dstillman** [comment]
How would general processing steps work? Isn't whatever we encode in the notes going to be dependent on the kinds of ids available in the export, which varies by translator?

> I think preserving annotation links is more than Zotero RDF can handle right now, especially since it doesn't get any annotation data yet 

I don't think it really needs annotation data — the `annotationKey` is included in the note metadata now but I don't think it really does much (@mrtcode can confirm), and we should probably strip it on export as long as we're embedding annotations in PDFs. So then we only need the item and attachment keys so that Show on Page and Show Item work.

**Comment by mrtcode** [comment]
> I don't think it really needs annotation data — the `annotationKey` is included in the note metadata now but I don't think it really does much (@mrtcode can confirm)

`annotationKey` is currently only used in Markdown translator.

Not sure if this is relevant for you, but we also store`attachmentURI` and `citationItem`.

**Comment by dstillman** [comment]
Yes, it's `attachmentURI` and `citationItem` that we'll need to update here.

So if we strip `annotationKey`, would the Markdown translator include the position instead? What happens now if you export to Markdown when the annotation item has been deleted?

**Comment by mrtcode** [comment]
> So if we strip `annotationKey`, would the Markdown translator include the position instead? What happens now if you export to Markdown when the annotation item has been deleted?

Markdown translator always includes page number and `annotationKey` (if it exists). So, if navigation by using `annotationKey` fails for any reason, then the page number is used.
